### PR TITLE
PERF-3594 Support auto tasks for workloads in nested directories

### PR DESCRIPTION
**Pull request by Fefer-Ivan**

### CODE REVIEW

**Comment by Fefer-Ivan** [comment]
https://spruce.mongodb.com/version/6387528361837d151c52e069/tasks

**Review by rtimmons — APPROVED** [review]

**Comment by Fefer-Ivan** [comment]
https://spruce.mongodb.com/version/63878a590305b95f2fad966b/tasks

**Review by rtimmons — DISMISSED** [review]
Nice little change. Just a couple comments to make it friendlier for future maintainers. Thanks for digging in and doing this the "right" way :)

File: src/lamplib/src/genny/tasks/auto_tasks.py [hunk 1]
```diff
@@ -220,13 +220,24 @@ def __init__(self, workspace_root: str, file_path: str, is_modified: bool, reade
 
         self.auto_run_info = auto_run_info
 
-    @property
-    def file_base_name(self) -> str:
-        return str(os.path.basename(self.file_path).split(".yml")[0])
+    def _get_relative_path_from_src_workloads(self):
+        relative_path = self.file_path.split("src/workloads/")
+        if len(relative_path) != 2:
```
[the remark this answers is not in the document]
Comment (Fefer-Ivan) [reply src/lamplib/src/genny/tasks/auto_tasks.py:225]: Done

File: src/lamplib/src/genny/tasks/auto_tasks.py [hunk 2]
```diff
@@ -220,13 +220,24 @@ def __init__(self, workspace_root: str, file_path: str, is_modified: bool, reade
...
 
     @property
     def snake_case_base_name(self) -> str:
-        return self._to_snake_case(self.file_base_name)
+        # Workload path is workspace_root/src/*/src/workloads/**/*.yml.
+        # To create a base name we leave only **/*.yml and convert it to snake case. To preserve
+        # legacy names, we omit the first directory that was matched as part of **.
+        relative_path = self._get_relative_path_from_src_workloads()
+        relative_path = os.path.splitext(relative_path)[0]
+        relative_path = relative_path.split(os.sep)
```
[the remark this answers is not in the document]
Comment (Fefer-Ivan) [reply src/lamplib/src/genny/tasks/auto_tasks.py:245]: Done

**Comment by Fefer-Ivan** [comment]
https://spruce.mongodb.com/version/63889bf73627e0534c4c70bc/tasks

File: src/lamplib/src/genny/tasks/auto_tasks.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rtimmons) [reply src/lamplib/src/genny/tasks/auto_tasks.py:245]: 🌶️ This is great! 🙏 👍

**Review by rtimmons — APPROVED** [review]

**Pull request merged**